llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8024,7 +8024,7 @@ bool llvm::matchSimpleRecurrence(const PHINode *P, BinaryOperator *&BO,
   for (unsigned i = 0; i != 2; ++i) {
     Value *L = P->getIncomingValue(i);
     Value *R = P->getIncomingValue(!i);
-    Operator *LU = dyn_cast<Operator>(L);
+    auto *LU = dyn_cast<Instruction>(L);
```

Comment:
You could directly cast to `BinaryOperator` here and then save the `cast<BinaryOperator>(LU)` below.